java/arcs/android/storage/database/DatabaseImpl.kt

```diff
@@ -122,6 +130,38 @@ class DatabaseImpl(
         }
     }
 
+    /**
+     * Returns the ID for the given [StorageKey] if one already exists, otherwise creates a new one
+     * for it.
+     */
+    @VisibleForTesting
+    fun getStorageKeyId(storageKey: StorageKey): StorageKeyId {
+        storageKey.toString()
+        val content = ContentValues().apply {
+            put("storage_key", storageKey.toString())
+        }
+        return writableDatabase.insertWithOnConflict(
+            "storage_keys", null, content, SQLiteDatabase.CONFLICT_IGNORE
+        )
+    }
+
+    /**
+     * Returns a map of field name to ID for each field in the given schema [TypeId].
+     *
+     * Call [getSchemaTypeId] first to get the [TypeId].
+     */
+    @VisibleForTesting
+    fun getSchemaFieldIds(schemaTypeId: TypeId): Map<FieldName, FieldId> {
```

Comment:
Also here, a TODO to consider using an LRU cache would be good.

Reply:
Done